### PR TITLE
Add server "prefer" attribute to NTP module

### PR DESCRIPTION
**Pull request by mingjunzhang2019**
SUMMARY
This is to add NTP server "prefer" attribute to NTP module.

ISSUE TYPE
- Module Enhancement Pull Request

COMPONENT NAME
sonic_ntp

ADDITIONAL INFORMATION
- regression test log: 
[regression_test.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/9767805/regression_test.log)

- regression test result: 
[regression-2022-10-12-10-50-29.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/9767807/regression-2022-10-12-10-50-29.html.pdf)

### CODE REVIEW

File: tests/regression/roles/sonic_ntp/defaults/main.yml [hunk 1]
```diff
@@ -81,6 +81,7 @@ tests:
         - address: "{{ ntp_ip_server_3 }}"
           minpoll: 7
           maxpoll: 10
+          prefer: true
 
   - name: test_case_06
```
Comment (kerry-meyer) [inline tests/regression/roles/sonic_ntp/defaults/main.yml:86]: Please add a test case to change the value of the "prefer" bool. (Changing it from "true" to "false" is sufficient. This is to verify that the update logic in the config file correctly handles the case of changing this value on an already existing server configuration.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
This change set looks good: All "functional" code changes look fine.

I have posted only one comment, inline, in the "tests/regression/roles/sonic_ntp/defaults/main.yml" file. (This is regarding an additional test case to verify that changing the value of the "prefer" bool works correctly).

File: tests/regression/roles/sonic_ntp/defaults/main.yml (hunk 1 above)
Comment (mingjunzhang2019) [reply tests/regression/roles/sonic_ntp/defaults/main.yml:86]: Add more regression test cases. Thanks

Comment (kerry-meyer) [reply tests/regression/roles/sonic_ntp/defaults/main.yml:86]: The additional coverage and corresponding test result looks good.

Thanks for adding this case.

**Review by kerry-meyer — APPROVED** [review]
All changes and test results look good.

Approved.

**Pull request merged**